### PR TITLE
Powershell 7 support

### PR DESCRIPTION
**Pull request by normj**
*Description of changes:*
Update the existing PowerShell Support to target PowerShell 7.0 and .NET Core 3.1. 

Changes:
* Major version bump AWSLambdaPSCore to 2.0.0
* Target netcoreapp3.1 for building project
* Target dotnetcore3.1 Lambda runtime
* Switch all templates to use module AWS.Tools.* modules
* Update default PowerShell SDK NuGet package to be [7.0.0](https://www.nuget.org/packages/Microsoft.PowerShell.SDK/7.0.0)
* Amazon.Lambda.PowerShellHost NuGet package targets .NET Core 3.1
* Added Layers switch when publishing which forwards to Amazon.Lambda.Tools `--function-layers` switch.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: PowerShell/Module/Templates/Blueprints/S3EventToSQS/s3tosqs.ps1.txt [hunk 1]
```diff
@@ -6,16 +6,17 @@
 #
 # The last item in the PowerShell pipeline will be returned as the result of the Lambda function.
 #
-# To include PowerShell modules with your Lambda function, like the AWSPowerShell.NetCore module, add a "#Requires" statement
-# indicating the module and version.
+# To include PowerShell modules with your Lambda function, like the AWS.Tools.S3 module, add a "#Requires" statement
+# indicating the module and version. If using an AWS.Tools.* module the AWS.Tools.Common module must also be a required.
```
Comment (boblodgett) [inline PowerShell/Module/Templates/Blueprints/S3EventToSQS/s3tosqs.ps1.txt]: nit: "must also be a required." => "is also required.". Or at least drop the "a" before required. (Repeated on the other *.txt files as well)

File: Libraries/test/TestPowerShellFunctions/PowerShellScriptsAsFunctions/PowerShellScriptsAsFunctions.csproj [hunk 2]
```diff
@@ -21,12 +21,9 @@
     </Content>
   </ItemGroup>
   <ItemGroup>
-    <PackageReference Include="Amazon.Lambda.Core" Version="1.0.0" />
-    <PackageReference Include="Amazon.Lambda.Serialization.Json" Version="1.3.0" />
-    <PackageReference Include="Amazon.Lambda.TestUtilities" Version="1.0.0" />
-  </ItemGroup>
-  <ItemGroup>
-    <DotNetCliToolReference Include="Amazon.Lambda.Tools" Version="2.2.0" />
```
Comment (boblodgett) [inline Libraries/test/TestPowerShellFunctions/PowerShellScriptsAsFunctions/PowerShellScriptsAsFunctions.csproj]: Reason Amazon.Lambda.Tools is removed? Not used anymore?

File: Libraries/src/Amazon.Lambda.PowerShellHost/Amazon.Lambda.PowerShellHost.csproj [hunk 3]
```diff
@@ -3,18 +3,25 @@
...
+    <VersionPrefix>2.0.0</VersionPrefix>
     <AssemblyName>Amazon.Lambda.PowerShellHost</AssemblyName>
     <PackageId>Amazon.Lambda.PowerShellHost</PackageId>
     <PackageTags>AWS;Amazon;Lambda;PowerShell</PackageTags>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Amazon.Lambda.Core" Version="1.0.0" />
+    <PackageReference Include="Amazon.Lambda.Core" Version="1.1.0" />
     <PackageReference Include="Microsoft.PowerShell.SDK" Version="6.0.4" />
```
Comment (boblodgett) [inline Libraries/src/Amazon.Lambda.PowerShellHost/Amazon.Lambda.PowerShellHost.csproj]: Doesn't the ItemGroup conditions handle the ref for this only for netcoreapp2.1 below?

File: PowerShell/Module/Templates/Blueprints/S3EventToSQS/s3tosqs.ps1.txt (hunk 1 above)
Comment (normj) [reply PowerShell/Module/Templates/Blueprints/S3EventToSQS/s3tosqs.ps1.txt]: Fixed in all templates

File: Libraries/test/TestPowerShellFunctions/PowerShellScriptsAsFunctions/PowerShellScriptsAsFunctions.csproj (hunk 2 above)
Comment (normj) [reply Libraries/test/TestPowerShellFunctions/PowerShellScriptsAsFunctions/PowerShellScriptsAsFunctions.csproj]: Yes, it was never meant to be on this test project and it was using the old style of .NET tools before .NET Core Global Tools existed.

File: Libraries/src/Amazon.Lambda.PowerShellHost/Amazon.Lambda.PowerShellHost.csproj (hunk 3 above)
Comment (normj) [reply Libraries/src/Amazon.Lambda.PowerShellHost/Amazon.Lambda.PowerShellHost.csproj]: Fixed, surprised this didn't cause a compile error.

**Pull request merged**